src/main/java/io/vertx/circuitbreaker/impl/CircuitBreakerMetrics.java

```diff
@@ -242,4 +174,145 @@ private void addLatency(JsonObject json, Histogram histogram, String prefix) {
       .put("99.5", histogram.getValueAtPercentile(99.5))
       .put("100", histogram.getValueAtPercentile(100)));
   }
+
+  private static class RollingWindow {
+    private final Summary history;
+    private final Summary[] buckets;
+    private final long bucketSizeInNs;
+
+    RollingWindow(long windowSizeInMs, int numberOfBuckets) {
+      if (windowSizeInMs % numberOfBuckets != 0) {
+        throw new IllegalArgumentException("Window size should be divisible by number of buckets.");
+      }
+      this.buckets = new Summary[numberOfBuckets];
+      for (int i = 0; i < buckets.length; i++) {
+        this.buckets[i] = new Summary();
+      }
+      this.bucketSizeInNs = 1_000_000 * windowSizeInMs / numberOfBuckets;
+      this.history = new Summary();
+    }
+
+    public void add(Operation operation) {
+      getBucket(operation.end).add(operation);
+    }
+
+    public Summary totalSummary() {
+      Summary total = new Summary();
+
+      total.add(history);
+      total.add(windowSummary());
+
+      return total;
+    }
+
+    public Summary windowSummary() {
+      Summary window = new Summary(buckets[0].bucketIndex);
+      for (Summary bucket : buckets) {
+        window.add(bucket);
+      }
+
+      return window;
+    }
+
+    public void updateTime() {
+      getBucket(System.nanoTime());
+    }
+
+    private Summary getBucket(long timeInNs) {
+      long bucketIndex = timeInNs / bucketSizeInNs;
+
+      //sample too old:
+      if (bucketIndex < buckets[0].bucketIndex) {
+        return history;
+      }
+
+      shiftIfNecessary(bucketIndex);
+
+      return buckets[(int) (bucketIndex - buckets[0].bucketIndex)];
+    }
+
+    private void shiftIfNecessary(long bucketIndex) {
+      long shiftUnlimited = bucketIndex - buckets[buckets.length - 1].bucketIndex;
+      if (shiftUnlimited <= 0) {
+        return;
+      }
+      int shift = (int) Long.min(buckets.length, shiftUnlimited);
+
+      // Add old buckets to history
+      for(int i = 0; i < shift; i++) {
+        history.add(buckets[i]);
+      }
+
+      System.arraycopy(buckets, shift, buckets, 0, buckets.length - shift);
+
+      for(int i = buckets.length - shift; i < buckets.length; i++) {
+        buckets[i] = new Summary(bucketIndex + i + 1 - buckets.length);
+      }
+    }
+
+    public long getMetricRollingWindowSizeInMs() {
+      return bucketSizeInNs * buckets.length / 1_000_000;
+    }
+
+    private static class Summary {
+      final long bucketIndex;
+      final Histogram statistics;
+
+      private int successes;
+      private int failures;
+      private int exceptions;
+      private int timeouts;
+      private int fallbackSuccess;
+      private int fallbackFailure;
+      private int shortCircuited;
+
+      private Summary() {
+        this(-1);
+      }
+
+      private Summary(long bucketIndex) {
+        this.bucketIndex = bucketIndex;
+        statistics = new Histogram(2);
```

Comment:
Propose to use 2 significant digits to use less space.  (4 kilobyte/histogram <> 64kilobyte per histogram)

No problem for me to change it back to 3.